### DISCUSSION TITLE
redux store external dependency needs window

### DISCUSSION THREAD

**Discussion by dsauna** [vercel/next.js#14992] [Help]
I'm using redux with the newest version of nextjs and typescript. My store initializes an external sdk to set some thunk values. The problem is that said sdk depends on the window object so I was originally getting the window undefined error. 

I fixed that by dynamially importing the sdk, but now I get a type error saying that the sdk method is not a constructor. Any help would be greatly appreciated

```
const sdk = dynamic(() => import('sdk'), { ssr: false }) as any;

const initialized = new sdk.initialize();

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(thunk.withExtraArgument({sdk)))
);

```

**Comment by jamesmosier** [comment]
`next/dynamic` only works with importing components, not libraries. 

`next/dynamic` is a fork of [react-loadable](https://github.com/jamiebuilds/react-loadable) which is meant for components.

**Reply by dsauna** [reply]
Thank you @jamesmosier! any idea what might solve my problem

**Reply by jamesmosier** [reply]
You will either have to wait to initialize the sdk until `typeof window !== 'undefined'` or remove `window` from the sdk itself
